### ISSUE TITLE
folder logic FULL vs PREFIX

### ISSUE THREAD

**Issue by larshp** [SAP2Moose/SAP2Moose#78]
Hi,
Using FULL folder logic forces contributors to install to that exact package name, eg. Z2MSE_DEMO which typically will be a transportable package. 
So it is not possible to install to a $ package and not putting the objects in a transport.

You can consider if you like to use the PREFIX folder logic, so instead of Z2MSE_DEMO it can be DEMO. Note that this requires the parent package to be a prefix of the child. Eg. Z2MSE_MAIN would have child package Z2MSE_MAIN_FAMIX instead of the current Z2MSE_FAMIX

For more info see:
http://docs.abapgit.org/ref-dot-abapgit.html

**Comment by RainerWinkler** [comment]
@larshp I will do it, thanks for proposing.
What is the safest/best way to migrate?
1. Create new packages (Maybe except for the top package) that are compatible to the PREFIX folder logic.
2. Move all elements to the new packages.
3. Delete all now obsolete packages.
4. Change the type of the repository to prefix.

In the description you linked, the top package is ZFOO, but the folder structure is described to start with /foo. Is there a logic to ignore a leading Z or Y? Or is it only a typing error?

**Comment by larshp** [comment]
its a bug in the doucmentation, fixed in https://github.com/larshp/abapGit/commit/ed12bcd3fbf6697a72ee191f4956ea235f7a6b99, thanks 😄

**Comment by RainerWinkler** [comment]
So I will do it as proposed in my four steps?

**Comment by larshp** [comment]
yes, abapGit might show some warnings when doing it, but just ignore these

**Comment by RainerWinkler** [comment]
Changes to packages:
Z2MSE -> unchanged
Z2MSE_DEMO -> unchanged
Z2MSE_MAIN -> unchanged
Z2MSE_EXTRACT3 -> Z2MSE_MAIN_EXTR
Z2MSE_ASSOCIATIONS -> Z2MSE_MAIN_EXTR_ASSOCIATIONS
Z2MSE_ASSOCIATION_BUILDERS -> Z2MSE_MAIN_EXTR_ASSOCBUILD
Z2MSE_ELEMENTS -> Z2MSE_MAIN_EXTR_ELEMENTS
Z2MSE_ELEMENT_BUILDERS -> Z2MSE_MAIN_EXTR_ELEMENTBUILD
Z2MSE_FAMIX -> Z2MSE_MAIN_FAMIX
Z2MSE_MSE -> Z2MSE_MAIN_MSE
Z2MSE_OBSOLETE -> Unchanged
Z2MSE_EXTRACT -> Z2MSE_OBSOLETE_EXTRACT
Z2MSE_SAP -> Z2MSE_OBSOLETE_SAP
Z2MSE_TEMPLATE -> Unchanged
Z2MSE_TEST -> Unchanged
Z2MSE_TEST_INITIAL_SELECTION -> Unchanged
Z2MSE_TEST_NO_INITIAL_SELECTN -> Unchanged
Z2MSE_TOOLS -> Unchanged

**Comment by RainerWinkler** [comment]
The files for the classes exist after step 2 now twice. Once in the old and once in the new folder.

**Comment by RainerWinkler** [comment]
@larshp 
There might be now obsolete folders and xml files in the system.
There is a new folder main, but also a folder z2mse_main, it contains a file package.devc.xml with this content:
<?xml version="1.0" encoding="utf-8"?>
<abapGit version="v1.0.0" serializer="LCL_OBJECT_DEVC" serializer_version="v1.0.0">
 <asx:abap xmlns:asx="http://www.sap.com/abapxml" version="1.0">
  <asx:values>
   <DEVC>
    <CTEXT>Z2MSE_Z2MSE_MAIN</CTEXT>
   </DEVC>
  </asx:values>
 </asx:abap>
</abapGit>

Are there still some cleanups required?

**Comment by larshp** [comment]
yeah, delete the old folders, eg https://github.com/SAP2Moose/SAP2Moose/tree/master/src/z2mse_demo note that since this is most likely a transportable package, its objects has to be released in a transport before the Z package can be deleted.

Alternatively, just delete the Z folders with a normal git client, or the github web interface

**Comment by larshp** [comment]
perfect now 👍

**Comment by RainerWinkler** [comment]
Close only after:
1. Documentation if abapGit is checked for possible improvements
2. Changelog is updated
